design/FY2017/MultipleAirloopsInOneZone.md

```diff
@@ -0,0 +1,546 @@
+
+
+### 6. Revise other places that assume a single airloop is associated with a zone
+Searching on `.ReturnAirNode` and `.AirLoopNum` shows relevant hits in: 
+
+ - AirflowNetworkBalanceManager.cc
+ - DualDuct.cc
+ - Furnaces.cc
+ - HVACFourPipeBeam.cc
+ - HVACManager.cc
+ - HVACMultiSpeedHeatPump.cc
+ - HVACUnitarySystem.cc
+ - MixedAir.cc
+ - PackagedTerminalHeatPump.cc
+ - PurchasedAirManager.cc
+ - ReportSizingManager.cc
+ - RoomAirModelAirflowNetwork.cc
+ - RoomAirModelManager.cc
+ - RoomAirModelUserTempPattern.cc
+ - SimAirServingZones.cc
+ - SimulationManager.cc
+ - SingleDuct.cc
+ - SystemAvailabilityManager.cc
+ - SystemReports.cc
+ - ZoneAirLoopEquipmentManager.cc
+ - ZoneContaminantPredictorCorrector.cc
+ - ZoneEquipmentManager.cc
+ - ZonePlenum.cc
+ - ZoneTempPredictorCorrector.cc
+ - And numerous unit tests
+
+### 7. Allow different sizing specifications for different air terminal units
+
+#### DataSizing::TermUnitFinalZoneSizing ####
+
+- Terminal units get their final sizing data from `TermUnitFinalZoneSizing`
+- TermUnitFinalZoneSizing` holds flow rates and other data which have been adjusted for sizing on ventilation load or other Sizing:System inputs such as a hard size for total air loop flow rate.
+- `TermUnitFinalZoneSizing` is initialized to `DataSizing::FinalZoneSizing` in  `ZoneEquipmentManager:UpdateZoneSizing`
+- `TermUnitFinalZoneSizing` is updated to adjusted `FinalZoneSizing` data in `SimAirServingZones::UpdateSysSizing`
+- `TermUnitFinalZoneSizing` is further adjusted in `SimAirServingZones::UpdateSysSizing`
+
+#### DataSizing::TermUnitSizing ####
+
+- `TermUnitSizing` is used as input in `SimAirServingZones::UpdateSysSizing` and in `ReportSizingManager::RequestSizing`.
+- `DataSizing::TermUnitSizing` is initialized in the various terminal unit sizing routines.
+
+#### SimAirServingZones::UpdateSysSizing ####
+- `UpdateSysSizing` uses `TermUnitSizing`, `ZoneSizing` and `FinalZoneSizing` as inputs.
+- `UpdateSysSizing` is basically four functions in one, split up by a big case statement for `CallIndicator` = BeginDay, DuringDay, EndDay, or EndSysSizingCalc (tempted to split that into four separate functions).
+- Lots of sizing calcs are done here, and the most pertinent section is that it makes final adjustments to `TermUnitFinalZoneSizing`.
+
+#### Proposed Changes ####
+- Add new struct `DataSizing::AirTerminalSizingData` to hold input for the new `DesignSpecification:AirTerminal:Sizing` object. (Similar to ZoneHVACSizingData)
+- The new array will be called `AirTerminalSizingSpecifications`.
+- Add new function `SizingManager::GetAirTerminalSizing` to process input for the new ` DesignSpecification:AirTerminal:Sizing` object.
+
+- Change `TermUnitSizing` and `TermUnitFinalZoneSizing` to be indexed by air distribution unit (ADU) number instead of by zone. Direct air terminal units will be added to the end of the ADU list for this indexing.
```

Comment:
Documenting thought...
This seems messy. If you can pre-calculate fractional values for these indexed arrays (using air node fraction * zone level value), then when the air loop sums the specific flows it should also be possible to use air node fraction * zone level value. During sizing you only know CurZoneEqNum (in RequestSizing) but would have to know which terminal unit is sizing so you could still use air node fraction * zone level value in the sizing arrays (or pass the correct air volume flow rate in a Data* var). It would be easier to start with this method and index the arrays later if you run into problems (fairly easy, yet tedious, to index these arrays).